frame/staking/src/lib.rs

```diff
@@ -1059,28 +1074,46 @@ decl_storage! {
 		/// The earliest era for which we have a pending, unapplied slash.
 		EarliestUnappliedSlash: Option<EraIndex>;
 
+		/// The last planned session scheduled by the session pallet.
+		///
+		/// This is basically in sync with the call to [`SessionManager::new_session`].
+		/// TODO: TWO_PHASE: needs care to set the initial value upon migration.
```

Comment:
Are there more notes about this? I assume a lot of chains will need to know what to do here. Also should this TODO be permanently stuck in this pallet?

Reply:
All the `TODO: TWO_PHASE` is intentional, and I will remove them once we finish the deployment (after this PR is merged and lives in kusama for a while, and once we switch staking to use two-phase).

Reply:
But I think this TODO is not really legit, the migration should give a correct session index here when this PR is shipped.

Generally speaking the TODO docs should just say about the two_phase unused result, and what will need to be removed once switched.

These doc goes on metadata and are shown on UI I think they need to be as precise as possible

Reply:
Good point, will tidy it up.

Reply:
Actually this is invalid If we do the two-phase migration (pun intended). The reason that I wrote it down was that this value is used to predict the election, will only be set when we plan an era, and the prediction will be inaccurate if it is not set it. 

It surely won't be a catastrophe if this is will be wrong until the first session, because we're not even using the output of the two-phase module.

Reply:
Yes ok, it seems that an invalid value here will just make `next_election_prediction` more in the future, which will not harm.